Bio-KBase-Auth/lib/Bio/KBase/AuthToken.pm

```diff
@@ -585,6 +592,54 @@ sub go_request {
 
 }
 
+sub _get_token {
+    my $self = shift @_;
+    my %p = @_;
+
+    my $json;
+    eval {
+	my $baseurl = $Bio::KBase::Auth::AuthSvcHost;
+#	my %headers;
+#        warn $p{'path'};
+	unless ($p{'path'}) {
+	    die "No path specified";
+	}
+#	$headers{'Content-Type'} = 'application/json';
+#	if (defined($p{'headers'})) {
+#	    %headers = (%headers, %{$p{'headers'}});
+#	}
+#	my $headers = HTTP::Headers->new( %headers);
+    
+#	my $client = LWP::UserAgent->new(default_headers => $headers);
+	my $client = LWP::UserAgent->new();
+	$client->timeout(5);
+	$client->ssl_opts(verify_hostname => 0);
+	my $method = $p{'method'} ? $p{'method'} : "POST";
+	my $url = sprintf('%s%s', $baseurl,$p{'path'});
+        $url=$p{'path'};
+        my $content={
+            'user_id'   =>  $p{'user_id'},
+            'password'  =>  $p{'password'},
+            'fields'    =>  'token',
+            };
+	my $response = $client->post($url, $content);
+#	if ($p{'body'}) {
+#	    $req->content( $p{'body'});
+#	}
+	unless ($response->is_success) {
+	    die $response->status_line;
+	}
+	$json = decode_json( $response->content());
+	$json = $self->_SquashJSONBool( $json);
```

Comment:
Probably not.  I want to read more on what SquashBool does before removing it though.